### PR TITLE
Add ExtractStatements to C API

### LINKED ISSUE #4977: Please add ExtractStatements to C API
Hi,

we're using the C API in the Go DuckDB driver. All statements are processed using the `duckdb_prepare` function. If someone provides multiple statements to our `Exec` function, DuckDB throws an error, complaining about multiple statements in a prepare, which makes absolutely sense. See https://github.com/marcboeker/go-duckdb/issues/46

The Python driver uses the [`ExtractStatements` method](https://github.com/duckdb/duckdb/blob/master/tools/pythonpkg/src/pyconnection.cpp#L145) on the provided input and then prepares each extracted statement on its own.

Would it be possible to add the [`ExtractStatements` method](https://github.com/duckdb/duckdb/blob/master/src/include/duckdb/main/connection.hpp#L103) to the C API, so that we can also accept multiple statements in one call to our `Exec` function.

Thank you!

### PR DESCRIPTION
**Pull request by LindsayWray**
Addresses issue #4977 

(Retry of a previous pr where I got myself in a merge mess with `master` and `feature`. )

### CODE REVIEW

File: src/main/capi/prepared-c.cpp [hunk 1]
```diff
@@ -14,6 +15,47 @@ using duckdb::QueryResultType;
...
+                                                duckdb_extracted_statements extracted_statements, idx_t index,
+                                                duckdb_prepared_statement *out_prepared_statement) {
+	if (!connection || !out_prepared_statement) {
+		return DuckDBError;
+	}
+	Connection *conn = (Connection *)connection;
+	auto source_wrapper = (ExtractStatementsWrapper *)extracted_statements;
+
+	auto wrapper = new PreparedStatementWrapper();
+	wrapper->statement = conn->Prepare(move(source_wrapper->statements[index]));
```
Comment (Mytherin) [inline src/main/capi/prepared-c.cpp:45]: Could we detect if `index` if out of bounds here, and add a test for that?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good - one more minor comment then it is good from my side:

File: src/main/capi/prepared-c.cpp (hunk 1 above)
Comment (LindsayWray) [reply src/main/capi/prepared-c.cpp:45]: ✔️

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**